flink-table/flink-table-planner/src/main/scala/org/apache/flink/table/planner/plan/rules/logical/LogicalCorrelateToJoinFromTemporalTableRule.scala

```diff
@@ -323,15 +306,47 @@ abstract class LogicalCorrelateToJoinFromGeneralTemporalTableRule(
     }
 
     val builder = call.builder()
-    val condition = builder.and(joinCondition, temporalCondition)
-
-    builder.push(leftInput)
-    builder.push(snapshot)
-    builder.join(correlate.getJoinType, condition)
-    val temporalJoin = builder.build()
+    val condition = builder.and(actualJoin.getCondition, temporalCondition)
+
+    val joinWithTemporalCondition = actualJoin.copy(
+      actualJoin.getTraitSet,
+      condition,
+      actualJoin.getLeft,
+      actualJoin.getRight,
+      actualJoin.getJoinType,
+      actualJoin.isSemiJoinDone)
+
+    val temporalJoin = if (actualJoin != rewriteJoin) {
+      rewriteJoin.replaceInput(0, joinWithTemporalCondition)
```

Comment:
Are there test cases which cover adding the projection?  

Also, does this run into any issues if there are multiple pushdowns?

Reply:
What cases do you have in mind?

Reply:
Do we have a test cases for both when the `actualJoin` is equal and not equal to the `rewriteJoin`?